src/DataReader/json_reader.rs

```diff
@@ -13,35 +13,35 @@ pub fn read_json<T: DeserializeOwned>(filename: String) -> Result<T> {
     let mut data = String::new();
     file.read_to_string(&mut data).unwrap();
 
-    let output: String = filename + ": Json format is not as expected";
-    let json_file: T = serde_json::from_str(&data).expect(output.as_str());
+    let json_file = serde_json::from_str(&data)
+        .unwrap_or_else(|_| panic!("{}: Json format is not as expected", filename));
```

Comment:
Is there a reason why unwrap_or_else combined with the panicking lambda  is preferred over using expect?

If its just to get the formatted error message couldnt the same have been acheived with `expect(format!())`?

Reply:
`expect(format!())` yields the same result, but this have the advantage of lazy evaluation